test/libsinsp_e2e/event_capture.h

```diff
@@ -32,184 +32,133 @@ limitations under the License.
 #include <functional>
 #include <mutex>
 #include <stdexcept>
+#include <utility>
+#include <sys/eventfd.h>
 
-class concurrent_object_handle_state_error : public std::logic_error {
-	using std::logic_error::logic_error;
-};
-
-class event_capture;
-
-/**
- * The concurrent_object_handle class encapsulates the task of accessing
- * event_capture::m_inspector in a thread-safe way, especially in the
- * run_callback_t functions passed to event_capture::run().
- */
-template<typename T>
-class concurrent_object_handle {
-public:
-	friend event_capture;
-
-	/**
-	 * Creates a new, unlocked handle with other's wrapped pointer and underlying mutex.
-	 * @param other
-	 */
-	concurrent_object_handle(const concurrent_object_handle& other) noexcept:
-	        m_object_ptr(other.m_object_ptr),
-	        m_object_lock(*other.m_object_lock.mutex(), std::defer_lock) {}
-
-	void lock() { m_object_lock.lock(); }
-
-	T* operator->() {
-		if(!m_object_lock.owns_lock()) {
-			throw concurrent_object_handle_state_error(
-			        "Attempt to access wrapped object without obtaining a lock.");
-		}
-		return m_object_ptr;
-	}
-
-	inline T* safe_ptr() { return operator->(); }
-
-	T& operator*() {
-		if(!m_object_lock.owns_lock()) {
-			throw concurrent_object_handle_state_error(
-			        "Attempt to access wrapped object without obtaining a lock.");
-		}
-		return *m_object_ptr;
-	}
-
-	T* unsafe_ptr() { return m_object_ptr; }
-
-	void unlock() { m_object_lock.unlock(); }
-
-private:
-	concurrent_object_handle(sinsp* object_ptr, std::mutex& object_mutex):
-	        m_object_ptr(object_ptr),
-	        m_object_lock(object_mutex, std::defer_lock) {}
-
-	T* m_object_ptr;
-	std::unique_lock<std::mutex> m_object_lock;
-};
+// Just a stupid fake FD value to signal to stop capturing events from driver and exit.
+// Note: we don't use it through eventfd because we want to make sure
+// that we received all events from the driver, until this very last close(FD_SIGNAL_STOP);
+#define EVENTFD_SIGNAL_STOP 1
```

Comment:
To avoid infinite loops when missing syscalls, after each test run() callback is called, we trigger the eventfd to signal that the capture must be stopped and all remaining events consumed.

Reply:
NOTE: i avoided sending a fake syscall event (like a `close(555)`) because if you are losing events, chances are high that you'll lose also this canary event, thus leading to infinitely looping.